Manually cancel intermediate draft builds

Ikuni17 opened the pull request.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/buildkite-pr-bot/builds/18)
* Commit: c92181f8d9127e259497737cfe44ed2a45756357

### Failed CI Steps
* [I am a failing step](https://buildkite.com/elastic/buildkite-pr-bot/builds/18#018c83e1-c444-4df3-a5b7-a4765cc1556b)

cc @Ikuni17

<!--buildkite-pr-comment-buildkite-pr-bot
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/buildkite-pr-bot/builds/18","number":18,"commit":"c92181f8d9127e259497737cfe44ed2a45756357"}],"number":18}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/buildkite-pr-bot/builds/19)
* Commit: 4560a8d818284adf52ff2e99dd728d7ae293f35d

### Failed CI Steps
* [I am a failing step](https://buildkite.com/elastic/buildkite-pr-bot/builds/19#018c8412-9796-490a-ab45-9004fd8c0d3c)

### History
* :broken_heart: [Build #18](https://buildkite.com/elastic/buildkite-pr-bot/builds/18) failed c92181f8d9127e259497737cfe44ed2a45756357

cc @Ikuni17

<!--buildkite-pr-comment-buildkite-pr-bot
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/buildkite-pr-bot/builds/19","number":19,"commit":"4560a8d818284adf52ff2e99dd728d7ae293f35d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/buildkite-pr-bot/builds/18","number":18,"commit":"c92181f8d9127e259497737cfe44ed2a45756357"}],"number":19}
buildkite-pr-comment-->

brianseeders commented:
Interesting. Is there a specific reason you want the builds to be cancelled? As a developer, I would like the ability to continue pushing to a draft PR while a run is going. I would be surprised if it got cancelled.

For example, while my tests are running, I could make documentation updates.

Also, kibana has a feature that allows individual commits to skip ci based on changes since the last green build, and this kind of interferes with that.

Ikuni17 commented:
@brianseeders Thanks for the feedback.

> Interesting. Is there a specific reason you want the builds to be cancelled? As a developer, I would like the ability to continue pushing to a draft PR while a run is going. I would be surprised if it got cancelled.
> 
> For example, while my tests are running, I could make documentation updates.


My thoughts were that it's "wasted" CI minutes since the status of the older commit is not reported as the primary PR checks once a new commit is pushed, even if the new commit does not run CI. So, CI would need to run again at some point anyways. I hadn't considered doc updates and not sure if this situation applies to that.


> Also, kibana has a feature that allows individual commits to skip ci based on changes since the last green build, and this kind of interferes with that.


Do you mean `skip_ci_on_only_changed`, build reuse, or both?
